src/utils/common-utils.ts

```diff
@@ -7,6 +7,21 @@ export enum ResourceType {
     FONT = 'font'
 }
 
+/**
+ * A PerformanceEntry or RumEvent that is sourced from the PerformanceAPI
+ */
+export interface HasLatency {
+    startTime: DOMHighResTimeStamp;
+    duration: DOMHighResTimeStamp;
+}
+
+/**
+ * Creates key to link a RumEvent to the PerformanceEntry that it is sourced from
+ * e.g. performanceKey(ResourceEvent) === performanceKey(PerformanceResourceTiming)
+ */
+export const performanceKey = (details: HasLatency) =>
```

Comment:
Note: the general purpose of `performanceKey()` deserves some discussion.

All RumEvents that are sourced from PerformanceEntries share (or will share) fields duration and startTime. These fields are highly unique because they are [DOMHighResTimestamps](https://developer.mozilla.org/en-US/docs/Web/API/DOMHighResTimeStamp). 

This pattern can also used to link monkey patched HttpEvents to PerformanceResourceTiming entries for with `initatorType = Fetch || Xhr`

Reply:
Good discussion -- consider adding it to the method's JavaDoc comment for posterity.

> startTime or duration is zero... can happen when resources are cached

Does this happen in practice? Do we need some logic in place to prevent conflicts?

Reply:
Updated the comment. It was my initial worry that duration might be zero when resources are cached. But when attempting to produce this, I have not seen duration actually be zero. Also, startTime will still be extremely unique.